electionleaflets/apps/analysis/views.py

```diff
@@ -0,0 +1,17 @@
+from django_filters.views import FilterView
+from leaflets.filters import LeafletFilter
+from leaflets.models import Leaflet
+
+
+class TagLeafletList(FilterView):
+    template_name = "analysis/tag_leaflets.html"
+    filterset_class = LeafletFilter
+    paginate_by = 30
+
+    def get_queryset(self):
+        qs = Leaflet.objects.all()
+        if "/without_party/" in self.request.path:
+            qs = qs.filter(ynr_party_id="")
+        if "/without_people/" in self.request.path:
+            qs = qs.filter(person_ids=[])
```

Comment:
This is a bit icky. For example, it would make this class hard to test in isolation and brittle to URL changes. I'd probably make this two classes tbh.